front/components/vaults/VaultLayout.tsx

```diff
@@ -45,3 +79,85 @@ export function VaultLayout({
     </RootLayout>
   );
 }
+
+function VaultBreadCrumbs({
+  owner,
+  vault,
+  category,
+  dataSource,
+  dataSourceView,
+  parentId,
+}: {
+  owner: WorkspaceType;
+  vault: VaultType;
+  category?: DataSourceViewCategory;
+  dataSource?: DataSourceType;
+  dataSourceView?: DataSourceViewType;
+  parentId?: string;
+}) {
+  const { nodes } = useDataSourceContentNodes({
+    owner,
+    dataSource: dataSource,
+    internalIds: parentId ? [parentId] : [],
+  });
+
+  const { title: folderName, parentInternalId } = nodes[0] || {};
+
+  if (!category) {
+    return null;
+  }
+
+  const items: { label: string; [key: string]: any }[] = [
+    {
+      icon:
+        vault.kind === "global" ? (
+          <PlanetIcon className="text-brand" />
+        ) : (
+          <LockIcon className="text-brand" />
+        ),
+      label: vault.name,
+      href: `/w/${owner.sId}/data-sources/vaults/${vault.sId}`,
+    },
+    {
+      icon: CATEGORY_DETAILS[category].icon,
+      label: CATEGORY_DETAILS[category].label,
+      href: `/w/${owner.sId}/data-sources/vaults/${vault.sId}/categories/${category}`,
+    },
+  ];
+
+  if (dataSourceView) {
+    items.push({
+      icon: dataSourceView.connectorProvider ? (
+        React.createElement(
+          CONNECTOR_CONFIGURATIONS[dataSourceView.connectorProvider]
+            .logoComponent
+        )
+      ) : (
+        <FolderIcon className="text-brand" />
+      ),
+      label: dataSourceView.connectorProvider
```

Comment:
Same for this there is a helper somewhere.

Reply:
why would we need a helper here?

Reply:
Sorry duplicated comment. Let me resolve.